app/url/create-url/createUrlInteractor.js

```diff
@@ -18,10 +21,19 @@ class CreateUrlInteractor {
             return;
         }
 
-        const url = this.urlFactory.create({pasteId: request.pasteId, hash: request.hash});
-        await this.urlRepository.createHash(url);
+        const paste = await this.pasteRepository.findById({id: request.pasteId}); 
 
-        this.presenter.presentSuccess();
+        if (!paste) {
+            this.presenter.presentFailure(new NotFound(`Paste with id ${request.pasteId} doesn't exist`));
+            return;
+        }
+
+        const hash = this.jwt.sign({id: request.pasteId, visibility: paste._visibility}, process.env.SECRET_KEY, {expiresIn: +request.expiresAfterMs});
```

Comment:
We should think about syncronization between token and paste lifespans, to be sure, that token would not be "fresh" after paste got expired.
But for now, you can set DEFAULT_TOKEN_EXPIRES_AFTER_MS = 24 hours.

Reply:
And rename `id` -> `pasteId`